src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs

```diff
@@ -698,13 +699,18 @@ public IActionResult BuildTransaction([FromBody] BuildTransactionRequest request
                     }
                 }
 
+                var (opReturnRawData, isValid) = request.OpReturnDataIsHex ? TryGetHexValue(request.OpReturnData) : (null, true);
+                if (!isValid)
+                    return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, "Hex OpReturn error.", "The OpReturnData is set as a hex value and could not be decoded");
+
                 var context = new TransactionBuildContext(this.network)
                 {
                     AccountReference = new WalletAccountReference(request.WalletName, request.AccountName),
                     TransactionFee = string.IsNullOrEmpty(request.FeeAmount) ? null : Money.Parse(request.FeeAmount),
                     MinConfirmations = request.AllowUnconfirmed ? 0 : 1,
                     Shuffle = request.ShuffleOutputs ?? true, // We shuffle transaction outputs by default as it's better for anonymity.
                     OpReturnData = request.OpReturnData,
+                    OpReturnRawData = opReturnRawData,
```

Comment:
What is the difference between OpReturnData and OpReturnRawData? Can't the consumer simply parse OpReturnData to Hex if they need to?

Reply:
The OpReturnData is an ASCII (String) and OpReturnRawData is Bytes

Reply:
It is the same data in different formats, I don't think the API should be responsible for returning the same data in multiple formats. Why can't the transformation happen by the API consumer?

Reply:
In this case you can send data to the API (for the OP_RETURN) in any format, HEX or ASCII..  more options (easy for the API consumer to use it)

p.e: you can easily send a hash like this, and will be 32 bytes and not 64 ASCII chars
35544c02f4cd4f403fab7579c6385400dbab656558b1d9750337ddb9f844e229

Reply:
Reviewed this and this is acceptable, see the other issue about duplicating the same logic in the fee calculation method.